src/debugAdapter/activateMplabxDebug.ts

```diff
@@ -153,19 +159,18 @@ async function convertDebugConfiguration(args: MplabxDebugConfiguration): Promis
 	// Find the elf
 	let outputFolder = path.join(args.program, 'dist', targetConfig.configurationName, args.debug ? 'debug' : 'production');
 
-	const fileType: string = '.elf';
+	const fileType: string = args.oldFileType ? '.cof' : '.elf';
```

Comment:
Is there a better way we can detect this? for instance, we could look for both the .cof and .elf file types all the time, or look for a specific type depending on the version of MPLABX

Reply:
Yes, I found that there is a variable called `DEBUGGABLE_SUFFIX` in `nbproject/Makefile-default.mk`. It will indicate the file extension name `cof` or `elf`, but I am not sure how to write it appropriately, so I simply use the settings to determine.
![image](https://github.com/user-attachments/assets/09200b1b-1e43-40a7-9fc0-94bd44247049)